api/helper.py

```diff
@@ -0,0 +1,210 @@
+from typing import List, Tuple, Type, TypeVar
+from datetime import datetime
+from werkzeug.datastructures import MultiDict
+from numpy import zeros, float32, array
+import numpy.typing as npt
+import re
+import json
+
+def parse_date(date_str: str) -> datetime:
+  """
+  Convert a string in the format `yyyy-mm-dd` or `yyyy/mm/dd` into a `datetime` object.
+
+  ### Parameters
+  - date_str : str
+    - The str to be converted.
+    - It must be in the format `yyyy-mm-dd` or `yyyy/mm/dd`.
+
+  ### Returns
+  - datetime
+    - The python `datetime` object represented by the string.
+
+  ### Raises
+  - ValueError* (Not implemented yet)
+    - If the date string is not of one of the two expected formats.
+  """
+  
+  parts = re.split("[/\\-]", date_str)
+  if len(parts) != 3:
+    raise RuntimeError("Invalid date format")
+  year = int(parts[0])  
+  month = int(parts[1])  
+  day = int(parts[2])  
+  return datetime(year, month, day)
+  
+def str_date(date: datetime) -> str:
+  """
+  Convert a `datetime` object into a string format for the polygon API.
+
+  ### Parameters
+  - date : datetime
+    - The date to be converted.
+
+  ### Returns
+  - str
+    - The string in the format `yyyy-mm-dd`.
+  """
+
+  return f"{date.year}-{str(date.month).zfill(2)}-{str(date.day).zfill(2)}"
+
+T = TypeVar('T')
+def get_param(args: MultiDict[str, str], key: str, name: str, type: Type[T]) -> T:
```

Comment:
new function, can get argument from HTTP params and cast to type all at the same time